testframework/src/main/java/net/neoforged/testframework/gametest/GameTestHelperFactory.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.testframework.gametest;
+
+import java.lang.invoke.MethodType;
+import java.util.function.Function;
+import net.minecraft.gametest.framework.GameTestHelper;
+import net.minecraft.gametest.framework.GameTestInfo;
+import net.neoforged.testframework.impl.ReflectionUtils;
+
+public interface GameTestHelperFactory<T extends GameTestHelper> extends Function<GameTestInfo, T> {
+    static <T extends GameTestHelper> GameTestHelperFactory<T> forType(Class<?> helperClass) {
+        var constructor = ReflectionUtils.constructor(helperClass, MethodType.methodType(void.class, GameTestInfo.class));
```

Comment:
Ideally the factories would be cached since this call can be expensive.  

You should also report the absence of the ctor concisely, rather than letting this just fail.

Reply:
Done. Please review this part as I have done it late and I don't trust my tired brain haha.